packages/form-core/src/validate/Validator.js

```diff
@@ -51,16 +64,17 @@ export class Validator {
 
   /**
    * @overridable
-   * @param {object} data
-   * @param {*} data.modelValue
-   * @param {string} data.fieldName
-   * @param {*} data.params
-   * @param {string} data.type
-   * @returns {string|Node|Promise<stringOrNode>|() => stringOrNode)}
+   * @param {object} [data]
+   * @param {*} [data.modelValue]
+   * @param {string} [data.fieldName]
+   * @param {*} [data.params]
+   * @param {string|undefined} [data.type]
+   * @returns {Promise<string|Node>}
```

Comment:
It can still return a string/Node withour promise?

Reply:
Since it's an async method, it always returns a promise no matter what. What is resolved is either a string or a Node. Or am I misunderstanding?

Reply:
You can determine whether it's async or not when you override it